src/formatter/formatter.go

```diff
@@ -2,20 +2,33 @@ package formatter
 
 import (
 	"fmt"
+	"strconv"
 	"strings"
 
 	"github.com/syucream/hakagi/src/constraint"
 )
 
 const (
-	baseSql = "ALTER TABLE %s ADD CONSTRAINT FOREIGN KEY (%s) REFERENCES %s(%s);"
+	baseSql   = "ALTER TABLE %s ADD CONSTRAINT FOREIGN KEY (%s) REFERENCES %s(%s);"
+	customSql = "ALTER TABLE %s ADD CONSTRAINT %s FOREIGN KEY (%s) REFERENCES %s(%s);"
 )
 
 func FormatSql(constraints []constraint.Constraint) string {
 	var queries []string
 
-	for _, c := range constraints {
-		q := fmt.Sprintf(baseSql, c.Table, c.Column, c.ReferedTable, c.ReferedColumn)
+	for i, c := range constraints {
+		var q string
+
+		// If the table name is too long, the foreign key name will also be too long,
+		// and an error will occur if the foreign key name exceeds 64 characters.
+		// I think 4 digits is enough for numbering.
+		if len(c.Table) >= 56 {
```

Comment:
64 characters - ( "\_ibfk\_"(6 characters) + number(1 or more characters) )
= 57 characters

Table name is allowed up to 57 characters, but considering the case where number is 2-digit, set it to 56 characters.